src/MeshKernelNET/Api/DisposableMesh2D.cs

```diff
@@ -152,12 +160,24 @@ public int NumNodes
             set { numNodes = value; }
         }
 
+        public int NumValidNodes
+        {
+            get { return numValidNodes; }
+            set { numValidNodes = value; }
+        }
+
         public int NumEdges
         {
             get { return numEdges; }
             set { numEdges = value; }
         }
 
+        public int NumValidEdges
```

Comment:
NumValidEdges should be a read-only property, it can be computed from the valid nodes.